packages/mask/src/plugins/Collectible/SNSAdaptor/MakeOfferDialog.tsx

```diff
@@ -114,14 +119,19 @@ export function MakeOfferDialog(props: MakeOfferDialogProps) {
     }, [open])
 
     const validationMessage = useMemo(() => {
-        const amount_ = new BigNumber(amount || '0')
+        const amountWithDecimal = Number(amount) * 10 ** Number(token.value?.decimals ?? 0)
```

Comment:
`import { pow10 } from '@masknet/web3-shared-base'`

Don't need a `amountWithDecimal`, calculate `amount` out using `Bignumber` methods rather than `Number`

Reply:
modified